The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9070916963/job/24923646840) [(plain)](https://github.com/rust-lang/rust/commit/28cb2d7dfb155ccb204f4d9a60af0e3460e59f5f/checks/24923646840/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling cfg-if v1.0.0
   Compiling adler v1.0.2
   Compiling rustc-demangle v0.1.24
   Compiling unwind v0.0.0 (/checkout/library/unwind)
error[E0599]: the function or associated item `allocator` exists for struct `Rc<T, A>`, but its trait bounds were not satisfied
     |
317  | / pub struct Rc<
317  | / pub struct Rc<
318  | |     T: ?Sized,
319  | |     #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global,
320  | | > {
     | |_- function or associated item `allocator` not found for this struct
...
1382 |           let alloc = unsafe { ptr::read(Self::allocator(&this)) };
     |                                                ^^^^^^^^^ function or associated item cannot be called on `Rc<T, A>` due to unsatisfied trait bounds
     |
note: if you're trying to build a new `Rc<T, A>` consider using one of the following associated functions:
      Rc::<T, A>::from_inner_in
      Rc::<T, A>::from_ptr_in
      Rc::<T, A>::new_in
      Rc::<T, A>::try_new_in
    --> library/alloc/src/rc.rs:375:5
     |
     |
375  |     unsafe fn from_inner_in(ptr: NonNull<RcBox<T>>, alloc: A) -> Self {
...
...
380  |     unsafe fn from_ptr_in(ptr: *mut RcBox<T>, alloc: A) -> Self {
...
...
694  |     pub fn new_in(value: T, alloc: A) -> Rc<T, A> {
...
...
797  |     pub fn try_new_in(value: T, alloc: A) -> Result<Self, AllocError> {
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: trait bound `T: Sized` was not satisfied
     |
     |
669  | impl<T, A: Allocator> Rc<T, A> {
     |      |
     |      unsatisfied trait bound introduced here
     |      unsatisfied trait bound introduced here
help: consider relaxing the type parameter's implicit `Sized` bound
     |
669  | impl<T: ?Sized, A: Allocator> Rc<T, A> {


error[E0599]: no method named `allocator` found for struct `ManuallyDrop<rc::Weak<T, A>>` in the current scope
     |
3064 |         let alloc = unsafe { ptr::read(this.allocator()) };
3064 |         let alloc = unsafe { ptr::read(this.allocator()) };
     |                                             ^^^^^^^^^ method not found in `ManuallyDrop<Weak<T, A>>`

error[E0599]: the function or associated item `allocator` exists for struct `Arc<T, A>`, but its trait bounds were not satisfied
     |
248  | / pub struct Arc<
248  | / pub struct Arc<
249  | |     T: ?Sized,
250  | |     #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global,
251  | | > {
     | |_- function or associated item `allocator` not found for this struct
...
1523 |           let alloc = unsafe { ptr::read(Self::allocator(&this)) };
     |                                                ^^^^^^^^^ function or associated item cannot be called on `Arc<T, A>` due to unsatisfied trait bounds
     |
note: if you're trying to build a new `Arc<T, A>` consider using one of the following associated functions:
      Arc::<T, A>::from_inner_in
      Arc::<T, A>::from_ptr_in
      Arc::<T, A>::new_in
      Arc::<T, A>::try_new_in
    --> library/alloc/src/sync.rs:289:5
     |
     |
289  |     unsafe fn from_inner_in(ptr: NonNull<ArcInner<T>>, alloc: A) -> Self {
...
...
294  |     unsafe fn from_ptr_in(ptr: *mut ArcInner<T>, alloc: A) -> Self {
...
...
711  |     pub fn new_in(data: T, alloc: A) -> Arc<T, A> {
...
...
844  |     pub fn try_new_in(data: T, alloc: A) -> Result<Arc<T, A>, AllocError> {
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: trait bound `T: Sized` was not satisfied
     |
     |
685  | impl<T, A: Allocator> Arc<T, A> {
     |      |
     |      unsatisfied trait bound introduced here
     |      unsatisfied trait bound introduced here
help: consider relaxing the type parameter's implicit `Sized` bound
     |
685  | impl<T: ?Sized, A: Allocator> Arc<T, A> {

error[E0599]: no function or associated item named `allocator` found for struct `sync::Weak` in the current scope
    --> library/alloc/src/sync.rs:2806:46
     |
     |
321  | / pub struct Weak<
322  | |     T: ?Sized,
323  | |     #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global,
324  | | > {
     | |_- function or associated item `allocator` not found for this struct
...
2806 |           let alloc = unsafe { ptr::read(Self::allocator(&this)) };
     |                                                ^^^^^^^^^ function or associated item not found in `Weak<T, A>`
     |
note: if you're trying to build a new `sync::Weak<T, A>` consider using one of the following associated functions:
      sync::Weak::<T, A>::new_in
      sync::Weak::<T, A>::from_raw_in
     |
     |
2627 |     pub fn new_in(alloc: A) -> Weak<T, A> {
...
...
2854 |     pub unsafe fn from_raw_in(ptr: *const T, alloc: A) -> Self {

For more information about this error, try `rustc --explain E0599`.
error: could not compile `alloc` (lib) due to 4 previous errors
warning: build failed, waiting for other jobs to finish...
```

</details>

--- a/library/alloc/src/rc.rs
+++ b/library/alloc/src/rc.rs
@@ -1379,7 +1379,7 @@ impl<T: ?Sized, A: Allocator> Rc<T, A> {
         let this = mem::ManuallyDrop::new(this);
         let ptr = Self::as_ptr(&this);
         // Safety: `this` is ManuallyDrop so the allocator will not be double-dropped
-        let alloc = unsafe { ptr::read(Self::allocator(&this)) };
+        let alloc = unsafe { ptr::read(&this.alloc) };
         (ptr, alloc)
     }
 
@@ -3061,7 +3061,7 @@ impl<T: ?Sized, A: Allocator> Weak<T, A> {
         let this = mem::ManuallyDrop::new(self);
         let result = this.as_ptr();
         // Safety: `this` is ManuallyDrop so the allocator will not be double-dropped
-        let alloc = unsafe { ptr::read(this.allocator()) };
+        let alloc = unsafe { ptr::read(&this.alloc) };
         (result, alloc)
     }
 

--- a/library/alloc/src/sync.rs
+++ b/library/alloc/src/sync.rs
@@ -1520,7 +1520,7 @@ impl<T: ?Sized, A: Allocator> Arc<T, A> {
         let this = mem::ManuallyDrop::new(this);
         let ptr = Self::as_ptr(&this);
         // Safety: `this` is ManuallyDrop so the allocator will not be double-dropped
-        let alloc = unsafe { ptr::read(Self::allocator(&this)) };
+        let alloc = unsafe { ptr::read(&this.alloc) };
         (ptr, alloc)
     }
 
@@ -2803,7 +2803,7 @@ impl<T: ?Sized, A: Allocator> Weak<T, A> {
         let this = mem::ManuallyDrop::new(self);
         let result = this.as_ptr();
         // Safety: `this` is ManuallyDrop so the allocator will not be double-dropped
-        let alloc = unsafe { ptr::read(Self::allocator(&this)) };
+        let alloc = unsafe { ptr::read(&this.alloc) };
         (result, alloc)
     }
 

